### DISCUSSION TITLE
Feature Request: Sum/Average over Batch

### DISCUSSION THREAD

**Discussion by DEKHTIARJonathan** [horovod/horovod#2754] [Q&A]
Hi,

As you may have seen, TensorFlow made default to aggregate gradients use sum across workers:

> Set default loss reduction as AUTO for improving reliability of loss scaling with distribution strategy and custom training loops. AUTO indicates that the reduction option will be determined by the usage context. For almost all cases this defaults to SUM_OVER_BATCH_SIZE. When used in distribution strategy scope, outside of built-in training loops such as tf.keras compile and fit, we expect reduction value to be 'None' or 'SUM'. Using other values will raise an error. 

Source: https://github.com/tensorflow/tensorflow/releases/tag/v1.14.0

That would be nice to have a flag switching for one to another. If I'm right the only way the change the default behavior for HVD is to re-install the lib with `HOROVOD_GPU_ALLREDUCE="NCCL"` or `HOROVOD_GPU_ALLGATHER="NCCL"`

Is this a feature that may be possible ?
